Core/src/Detector/CylindricalContainerBuilder.cpp

```diff
@@ -0,0 +1,149 @@
+/// @param gctx The geometry context of the this call
+/// @param objects The object vector
+/// @param binnning the chosen binning
+/// @param logLevel the logging output level
+///
+/// @note no checking on consistency is done as the caller container builder
+/// is already checked at construction
+///
+/// @return a newly built container
+template <typename object_collection>
+Acts::Experimental::DetectorComponent::Container connect(
+    const Acts::GeometryContext& gctx, object_collection& objects,
+    const std::vector<Acts::BinningValue>& binning,
+    Acts::Logging::Level logLevel) {
+  // Return container object
+  Acts::Experimental::DetectorComponent::Container rContainer;
+  if (binning.size() == 1u) {
+    Acts::BinningValue bv = binning.front();
+    // 1-dimensional binning options
+    switch (bv) {
+      case Acts::binR: {
+        rContainer =
+            Acts::Experimental::detail::CylindricalDetectorHelper::connectInR(
+                gctx, objects, {}, logLevel);
+      } break;
+      case Acts::binZ: {
+        rContainer =
+            Acts::Experimental::detail::CylindricalDetectorHelper::connectInZ(
+                gctx, objects, {}, logLevel);
+      } break;
+      case Acts::binPhi: {
+        rContainer =
+            Acts::Experimental::detail::CylindricalDetectorHelper::connectInPhi(
+                gctx, objects, {}, logLevel);
+      } break;
+      default:
+        break;
+    }
+  } else if (binning ==
+                 std::vector<Acts::BinningValue>{Acts::binZ, Acts::binR} and
+             objects.size() == 2u) {
+    rContainer =
+        Acts::Experimental::detail::CylindricalDetectorHelper::wrapInZR(
+            gctx, objects, logLevel);
+  }
+  return rContainer;
+}
+}  // namespace
+
+Acts::Experimental::CylindricalContainerBuilder::CylindricalContainerBuilder(
+    const Acts::Experimental::CylindricalContainerBuilder::Config& cfg,
+    std::unique_ptr<const Acts::Logger> logger)
+    : IDetectorComponentBuilder(), m_cfg(cfg), m_logger(std::move(logger)) {
+  // Check if builders are present
+  if (m_cfg.builders.empty()) {
+    throw std::invalid_argument(
+        "CylindricalContainerBuilder: no sub builders provided.");
+  }
+  // Check if binning value is correctly chosen
+  if (m_cfg.binning.size() == 1u) {
+    // 1-dimensional case
+    auto b = m_cfg.binning.front();
+    if (b != Acts::binR and b != Acts::binZ and b != Acts::binPhi) {
+      throw std::invalid_argument(
+          "CylindricalContainerBuilder: 1D binning only supported in z, r, or "
+          "phi");
+    }
+  } else if (m_cfg.binning.size() == 2u) {
+    // 2-dimensional case, this is for wrapping
+    if (m_cfg.binning !=
+        std::vector<Acts::BinningValue>{Acts::binZ, Acts::binR}) {
+      throw std::invalid_argument(
+          "CylindricalContainerBuilder: 2D binning only supports wrapping in "
+          "z-r.");
+    } else if (m_cfg.builders.size() != 2u) {
```

Comment:
If this check runs before the above function, then neglect my comment above :P

Reply:
yes, you won't be able to construct a different object as it will throw an exception.